lib/Sema/CSSimplify.cpp

```diff
@@ -2972,6 +2972,11 @@ performMemberLookup(ConstraintKind constraintKind, DeclName memberName,
       
     } else {
       if (!hasStaticMembers) {
+        if (cand->isImplicit() && isa<ConstructorDecl>(cand)) {
```

Comment:
Constructor search fails when you move this check outside this case, but not here.  If we've already got here, then we know that the lookup is bogus.  This just excludes the extra diagnostic caused by that kind of invalid protocol conformance.